src/logging/Manager.cc

```diff
@@ -1731,6 +1731,79 @@ WriterFrontend* Manager::CreateWriterForFilter(Filter* filter, const std::string
                         from_remote, filter->name);
 }
 
+bool Manager::WritesFromRemote(const detail::LogWriteHeader& header, std::vector<detail::LogRecord>&& arg_records) {
+    auto records = std::move(arg_records);
```

Comment:
I'm confused by the arguments here. I understand the header being const-reference since you're only using the values out of it and not copying it. The rvalue-reference for arg_records doesn't make as much sense. If you're going to immediately `std::move` it, why not make it pass-by-value and have the caller std::move into it? Then you don't need the extra `std::move` at the start of the method.

Reply:
I've used the rvalue here as the method is intended to be a sink and taking ownership, plus preventing the caller from mistakenly creating a copy (which for a batch of log writes might be significant).

Elsewhere there exists the following which broker currently calls - it's somewhat consistent with that also (though I was the one that added this)
```
bool WriteFromRemote(EnumVal* id, EnumVal* writer, const std::string& path, detail::LogRecord&& rec);
```

> Then you don't need the extra std::move at the start of the method.

I only added this one as sonarlint warned that I never moved `arg_records` so I added that one. I can also just remove it assuming nothing else warns.

The actual move happens per record further down:
```
    for ( auto& r : records )
        w->second->writer->Write(std::move(r));
```

I like the rvalue to indicate the passing of ownership, but if you think that's quirky, maybe we can change both APIs as a follow-up?

Reply:
> I only added this one as sonarlint warned that I never moved arg_records so I added that one. I can also just remove it assuming nothing else warns.

I've removed the `auto record = std::move(...)` for now.

Reply:
> I like the rvalue to indicate the passing of ownership, but if you think that's quirky, maybe we can change both APIs as a follow-up?

I think this is fine. Maybe a note on the function comment indicating that that's the case.